modules/simple_forms_api/app/services/simple_forms_api/s3/submission_archive_builder.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+require 'csv'
+require 'fileutils'
+
+# built in accordance with the following documentation:
+# https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/platform/practices/zero-silent-failures/remediation.md
+module SimpleFormsApi
+  module S3
+    class SubmissionArchiveBuilder < Utils
+      def initialize(benefits_intake_uuid: nil, submission: nil, **options) # rubocop:disable Lint/MissingSuper
+        defaults = default_options.merge(options)
+
+        @submission = submission || FormSubmission.find_by(benefits_intake_uuid:)
+        raise 'Submission was not found' unless @submission
+
+        @benefits_intake_uuid = @submission.benefits_intake_uuid
+
+        assign_instance_variables(defaults)
+      end
+
+      def run
+        FileUtils.mkdir_p(temp_directory_path)
+
+        process_submission_files
+
+        temp_directory_path
+      rescue => e
+        handle_error("Failed building submission: #{submission.id}", e, { benefits_intake_uuid: })
+      end
+
+      private
+
+      attr_reader :attachments, :benefits_intake_uuid, :file_path, :include_json_archive, :include_manifest,
+                  :include_text_archive, :metadata, :parent_dir, :submission
+
+      def default_options
+        {
+          attachments: [], # an array of attachment confirmation codes
+          file_path: nil, # file path for the PDF file to be archived
+          include_json_archive: true, # include the form data as a JSON object
+          include_manifest: true, # include a CSV file containing manifest data
+          include_text_archive: true, # include the form data as a text file
+          metadata: {}, # pertinent metadata for original file upload/submission
+          parent_dir: 'vff-simple-forms' # S3 bucket base directory where files live
+        }
+      end
+
+      def process_submission_files
+        write_pdf
+        write_as_json_archive if include_json_archive
+        write_as_text_archive if include_text_archive
+        write_attachments unless attachments.empty?
+        write_manifest if include_manifest
+        write_metadata
+      end
+
+      def write_pdf
+        write_tempfile(submission_pdf_filename, File.read(generate_pdf_content))
+      end
+
+      # TODO: this will be pulled out to be more team agnostic
+      def generate_pdf_content
+        return file_path if file_path
+
+        form_number = SimpleFormsApi::V1::UploadsController::FORM_NUMBER_MAP[submission.form_type]
+        form = "SimpleFormsApi::#{form_number.titleize.gsub(' ', '')}".constantize.new(form_data_hash)
+        filler = SimpleFormsApi::PdfFiller.new(form_number:, form:)
+
+        @file_path = filler.generate(timestamp: submission.created_at)
+        @metadata = SimpleFormsApiSubmission::MetadataValidator.validate(
```

Comment:
In the context here I might pull out the creation of `@metadata` since we don't actually need it later in this method. It also gets used outside of here (like in `write_metadata`) so maybe that's a more appropriate place for it? Just a thought!

Reply:
I also considered that but I didn't want to make form another instance variable. I went ahead and refactored this area, it doesn't quite move the metadata logic to that method but I think it does break things up a bit better.